.github/workflows/tidy-modern.yml

```diff
@@ -0,0 +1,34 @@
+name: tidy-modern
```

Comment:
How about renaming this (the yml-file as well) to clang-tidy? "modern" implies that this runs only the "modernize" checks, but in the future we probably also want to run other checks from clang-tidy